contracts/core/management/TokenExclusionManager.sol

```diff
@@ -316,34 +313,32 @@ contract TokenExclusionManager is
   /**
    * @notice Attempts to transfer the user's share of a removed token to them.
    * @dev Calculates the user's share based on their portfolio token balance at the time of removal and transfers it.
-   * @param _removedToken Address of the removed token.
    * @param _user address of user to claim removed token
+   * @param _snapshotId The ID of the snapshot at which the token was removed.
    * @param _portfolioTokenBalance User's balance of portfolio tokens at the last valid snapshot ID.
-   * @param _totalSupply Total supply of portfolio tokens at the snapshot ID.
    */
   function attemptClaim(
-    address _removedToken,
     address _user,
     uint256 _snapshotId,
-    uint256 _portfolioTokenBalance,
-    uint256 _totalSupply
+    uint256 _portfolioTokenBalance
   ) private {
     RemovedTokenRecord memory tokenInformation = removedToken[_snapshotId];
     // Calculate the user's share of the removed token
-    uint256 currentVaultBalance = IERC20Upgradeable(tokenInformation.token)
+    uint256 totalSupply = tokenInformation.totalSupply;
+    address currentRemovedToken = tokenInformation.token;
+
+    uint256 currentVaultBalance = IERC20Upgradeable(currentRemovedToken)
       .balanceOf(tokenInformation.vault);
     uint256 balance = (currentVaultBalance * _portfolioTokenBalance) /
-      _totalSupply;
+      totalSupply;
     // Transfer the calculated share to the user
     ITokenRemovalVault tokenRemovalVault = ITokenRemovalVault(
       tokenInformation.vault
     );
-    tokenRemovalVault.withdrawTokens(_removedToken, _user, balance);
+    tokenRemovalVault.withdrawTokens(currentRemovedToken, _user, balance);
 
     // Update the total supply record for the snapshot ID
-    removedToken[_snapshotId].totalSupply =
-      _totalSupply -
-      _portfolioTokenBalance;
+    totalSupply = totalSupply - _portfolioTokenBalance;
```

Comment:
Here this is not correct , we need - 
removedToken[_snapshotId].totalSupply =  _totalSupply -
      _portfolioTokenBalance; because we need to update the mapping